File: Tutoriels/Cpp/Co/code/src/manager/GSetting.c
//===============================================
#include "GSetting.h"
#include "GString.h"
#include "GConfig.h"
#include "GFree.h"
//===============================================
static GSettingO* m_GSettingO = 0;
//===============================================
const char G_CHAR_COMMENT = '#';
//===============================================
GSettingO* GSetting_Constructor();
void GSetting_Init(GSettingO* obj);
void GSetting_Load(const char* file);
//===============================================
GSettingO* GSetting_Constructor() {
    GSettingO* lObj = (GSettingO*)malloc(sizeof(GSettingO));
    GSetting_Init(lObj);
    return lObj; 
}
//===============================================
void GSetting_Init(GSettingO* obj) {
    obj->Load = GSetting_Load;
}
//===============================================
GSettingO* GSetting() {
    if(m_GSettingO == 0) {
        m_GSettingO = GSetting_Constructor();
    }
    return m_GSettingO;
}
//===============================================
void GSetting_Load(const char* file) {
    FILE *lFile = fopen(file, "r");
    char lBuffer[100];

    while(fgets(lBuffer, sizeof(lBuffer), lFile) != NULL) {
        char* lTrim = GString()->Trim(lBuffer);
        if(lTrim == 0) {continue;}
        char lFirst = lTrim[0];
        if(lFirst == G_CHAR_COMMENT) {GFree()->Free(lTrim); continue;}
        int lCount;
        char** lSplit = GString_Split(lTrim, "=", &lCount);
        char* lKey = GString()->Trim(lSplit[0]);
        char* lValue = GString()->Trim(lSplit[1]);
        GConfig()->Set_Data(lKey, lValue);
        printf("%s = %s\n", lKey, lValue);
        GFree()->Free(lTrim);
        GFree()->Free(lKey);
        GFree()->Free(lValue);
        GFree()->Free2(lSplit, lCount);
    }
    printf("\n");
    fclose(lFile);
}
//===============================================

File: php/class/GSetting.php
<?php   
    class GSetting {
        //===============================================
        private static $m_instance = null;
        //===============================================
        private function __construct() {
        
        }
        //===============================================
        public static function Instance() {
            if(is_null(self::$m_instance)) {
                self::$m_instance = new GSetting();  
            }
            return self::$m_instance;
        }
        //===============================================
        public function load($file) {
            GFile::Instance()->existData($file);
            $lFile = GFile::Instance()->getRealPath($file);
			$lFopen = fopen($lFile, "r");
			
			while(!feof($lFopen)) {
				$lLine = fgets($lFopen);
				$lLine = GString::Instance()->trimData($lLine);
				if($lLine == "") continue;
				if($lLine[0] == "#") continue;
				$lSplit = GString::Instance()->splitData($lLine);
				$lKey = $lSplit[0];
				$lValue = $lSplit[1];
				GConfig::Instance()->setData($lKey, $lValue);
			}
            $lConfigShow = GConfig::Instance()->getData("CONFIG_SHOW");
            if($lConfigShow == "TRUE") GConfig::Instance()->showData();
            fclose($lFopen);
        }
        //===============================================
    }
?>